tests/e2e_tests/spec_linespread_e2e.py

```diff
@@ -0,0 +1,304 @@
+import os
+import sys
+import glob
+import numpy as np
+import astropy.io.fits as fits
+from datetime import datetime, timedelta
+import logging
+import pytest
+import argparse
+import warnings
+from astropy.io.fits.verify import VerifyWarning
+
+from corgidrp.data import Dataset, DispersionModel, LineSpread
+from corgidrp.spec import compute_psf_centroid, calibrate_dispersion_model
+from corgidrp.data import Image
+from corgidrp.mocks import create_default_L2b_headers
+from corgidrp.walker import walk_corgidrp
+from corgidrp.check import (check_filename_convention, check_dimensions, 
+                           verify_hdu_count, verify_header_keywords, 
+                           get_latest_cal_file)
+
+
+
+# ================================================================================
+# Main Spec Linespread E2E Test Function
+# ================================================================================
+
+def run_spec_linespread_e2e_test(e2edata_path, e2eoutput_path):
+    """Run the complete spectroscopy linespread end-to-end test.
+    
+    This function consolidates all the test steps into a single linear flow
+    for easier reading and understanding.
+    
+    Args:
+        e2edata_path (str): Path to input data directory
+        e2eoutput_path (str): Path to output directory
+        
+    Returns:
+        tuple: (disp_model, coeffs, angle) from the baseline performance checks
+    """
+    
+    # ================================================================================
+    # (1) Setup Input Files
+    # ================================================================================
+    logger.info('='*80)
+    logger.info('Pre-test: set up input files and save to disk')
+    logger.info('='*80)
+
+    # Check if input folder already contains the expected files
+    existing_files = sorted(glob.glob(os.path.join(e2edata_path, 'cgi_*_l2b.fits')))
+    
+    if existing_files:
+        logger.info(f"Found {len(existing_files)} existing L2b files in {e2edata_path}")
+        logger.info("Using existing input files (skipping generation)")
+        saved_files = existing_files
+        l2b_dataset_with_filenames = Dataset(saved_files)
+    else:
+        logger.info("No existing input files found. Generating new input files...")
+        
+        # Load test data
+        datadir = os.path.join(os.path.dirname(__file__), '../test_data/spectroscopy')
+        file_path_science = os.path.join(datadir, "g0v_vmag6_spc-spec_band3_unocc_CFAM3_R1C2SLIT_PRISM3_offset_array.fits")
+        file_path_spot = os.path.join(datadir, "g0v_vmag6_spc-spec_band3d_unocc_CFAM3_R1C2SLIT_PRISM3_offset_array.fits")
```

Comment:
I haven't had a chance for a full review, but wanted to raise this first. In my folder I have `g0v_vmag6_spc-spec_band3_unocc_CFAM3d_R1C2SLIT_PRISM3_offset_array.fits` but not the file specified here.